File: 3/3.2.pas
program f1;

var  a, b, x, x_start, x_end, x_delta, iterator: Integer;

function f(x: Real): Real;
begin
  Result := Ln(1.2 * x + 2.3) + (Exp(3 * x) * Sinh(x));
end;

function g(x: Real): Real;
begin
  Result := Power(5, Power(x, 2) + 1) * Power(Power(x, 3.3) + Power(Abs(1 - x), 0.5), 0.5);
end;

function h(x: Real): Real;
begin
  Result := Power(1 + Power(Tan(x), 2), Arctan(x) + Ln(x) + 1);
end;

begin
  write('a = ');
  readln(a);
  write('b = ');
  readln(b);
  write('начало = ');
  readln(x_start);
  write('конец = ');
  readln(x_end);
  write('интервал = ');
  readln(x_delta);

  writeln('x', #9, '|', 'y');
  writeln('+-----------------------------+');

  // Способ 1
  iterator := 0;
  for x := x_start to x_end do
  begin
    if (iterator = 0) or (iterator mod x_delta = 0) then
      if x <= a then
        writeln(x, #9, '|', f(x))
      else if (a < x) and (x < b) then
        writeln(x, #9, '|', g(x))
      else if x >= b then
        writeln(x, #9, '|', h(x));
      iterator += 1
  end;

  writeln('+-----------------------------+');

  x := x_start;
  while x <= x_end do
  begin
    if x <= a then
      writeln(x, #9, '|', f(x))
    else if (a < x) and (x < b) then
      writeln(x, #9, '|', g(x))
    else if x >= b then
      writeln(x, #9, '|', h(x));
    x += x_delta;
  end;
  
  writeln('+-----------------------------+');
  
  //Способ 3
  x := x_start;
  repeat
    if x <= a then
      writeln(x, #9, '|', f(x))
    else if (a < x) and (x < b) then
      writeln(x, #9, '|', g(x))
    else if x >= b then
      writeln(x, #9, '|', h(x));
    x += x_delta;
  until x > x_end;
end.

File: 3/3.2.py
import math


def f(x: float) -> float:
    return math.log(1.2 * x + 2.3) + ((math.e ** (3 * x)) * math.sinh(x))


def g(x: float) -> float:
    return 5 ** (x ** 2 + 1) * (x ** 3.3 + (abs(1 - x)) ** 0.5) ** 0.5


def h(x: float) -> float:
    return (1 + math.tan(x) ** 2) ** (math.atan(x) + math.log10(x) + 1)


a = float(input("a = "))
b = float(input("b = "))
x_start = float(input("начало = "))
x_end = float(input("конец = "))
x_delta = float(input("интервал = "))


print(f"x\t|y")
print("+-----------------------------+")

# Способ 1
for x in range(int(x_start), int(x_end) + 1, int(x_delta)):
    if x <= a:
        print(f"{x}\t|{f(x)}")
    elif a < x < b:
        print(f"{x}\t|{g(x)}")
    elif x >= b:
        print(f"{x}\t|{h(x)}")

print("+-----------------------------+")

# Способ 2
while x_start <= x_end:
    if x_start <= a:
        print(f"{int(x_start)}\t|{f(x_start)}")
    elif a < x_start < b:
        print(f"{int(x_start)}\t|{g(x_start)}")
    elif x_start >= b:
        print(f"{int(x_start)}\t|{h(x_start)}")
    x_start += x_delta
